server/src/main/java/org/tctalent/server/request/candidate/PublishedDocBuilderServiceImpl.java

```
@@ -33,110 +40,187 @@
-      Map<String,CandidateProperty> properties = candidate.getCandidateProperties();
-      if (properties != null) {
-        //Fetch the value
-        CandidateProperty property = properties.get(propertyName);
-        if (property != null) {
-          val = property.getValue();
+        } else if (propertyName != null) {
+            //Check for the specific candidate property with the property name provided
+            Map<String, CandidateProperty> properties = candidate.getCandidateProperties();
+            if (properties != null) {
+                //Fetch the value
+                CandidateProperty property = properties.get(propertyName);
+                if (property != null) {
+                    val = property.getValue();
+                }
+            }
+        } else {
+            val = valueSource.getConstant();
         }
-      }
-    } else {
-      val = valueSource.getConstant();
+        return val;
     }
-    return val;
-  }
 
+    /**
+     * Retrieve the multi valued data value corresponding to the given PublishedDocColumnDef
+     *
+     * @param candidate Candidate
+     * @param columnDef Column definition
+     * @return the multi valued data or null if no such data is found.
+     */
+    @Nullable
+    private HasMultipleRows getExpandingData(
+        @NonNull Candidate candidate, @Nullable PublishedDocColumnDef columnDef) {
+        HasMultipleRows obj = null;
+        if (columnDef != null) {
+            final PublishedDocValueSource value = columnDef.getContent().getValue();
+            if (value != null) {
+                final Object o = fetchData(candidate, value);
+                if (o instanceof String stringVal) {
+                    try {
+                        JsonNode jsonNode = jsonObjectMapper.readTree(stringVal);
+                        obj = new JsonRows(jsonNode);
+                    } catch (JsonProcessingException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Swallowing exceptions without logging can make it difficult to identify and troubleshoot issues, potentially leading to application instability, hidden bugs, or unexpected behavior. To avoid this, ensure that all exceptions are logged using appropriate logging levels. When logging exceptions, include the exception object or its stack trace to capture detailed information. For critical exceptions, rethrow them when necessary to provide sufficient context for effective troubleshooting.